q2_quality_control/_evaluate_seqs.py

```diff
@@ -0,0 +1,76 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2017, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import tempfile
+import pandas as pd
+
+from ._utilities import _plot_histogram
+from ._blast import _blast, _perc_coverage, _run_command
+
+
+def _evaluate_seqs(query_sequences, reference_sequences, show_alignments):
+    cmd = _blast(query_sequences, reference_sequences, None, 0)
+    # modify command to output seq alignment information
+    cmd[6] = ('6 qseqid sseqid pident length mismatch gapopen qstart qend '
+              'sstart send evalue bitscore qlen qseq sseq')
+    # generate report of blast alignment results
+    results, alignments = _generate_alignments(cmd, show_alignments)
+
+    # plot histogram of mismatches between obs seqs and top hit in exp seqs
+    g = _plot_histogram(pd.to_numeric(results['Mismatches']))
+
+    return results, alignments, g
+
+
+def _generate_alignments(cmd, show_alignments):
+    '''Run command line subprocess and extract hits.'''
+    with tempfile.NamedTemporaryFile() as output:
+        cmd = cmd + [output.name]
+        _run_command(cmd)
+        return _generate_alignment_results(output.name, show_alignments)
+
+
+def _generate_alignment_results(blast_results, show_alignments):
+    '''blast_results: output of blastn in outfmt == 6'''
+    alignments = []
+    results = []
+    with open(blast_results, "r") as inputfile:
```

Comment:
Are `blast` results always TSV? It makes me a little nervous to re-implement a TSV parser here. Maybe this could use pandas, or python's [`csv`](https://docs.python.org/3.5/library/csv.html) instead?

Reply:
👍 pandas makes sense since the outputs are dataframes